pkg/cli/debug/debug.go

```diff
@@ -1176,6 +1177,73 @@ func (o *DebugOptions) getLogs(pod *corev1.Pod) error {
 	}.RunLogs()
 }
 
+// getNamespace returns namespace name and clean up function.
+// --to-namespace flag has the highest priority. If it is not set, -n flag is used.
+// if there is no explicit namespace flag(--to-namespace, -n), default one is used.
+// In the manner of node debugging, if default namespace is decided to be used and
+// this namespace is not privileged, this function creates temporary namespace.
+func (o *DebugOptions) getNamespace(infoNs string) (string, func(), error) {
+	if len(o.ToNamespace) > 0 {
+		return o.ToNamespace, func() {}, nil
+	}
+
+	if len(infoNs) == 0 {
+		infoNs = o.Namespace
+	}
+
+	if o.ExplicitNamespace {
+		return infoNs, func() {}, nil
+	}
+
+	if o.IsNode {
```

Comment:
Nit:
reverse this condition to be:
```golang
if !o.IsNode {
    return infoNs, func() {}, nil
}
// the rest goes here
```
this will make the code easier to read and understand.